rasa/utils/tensorflow/models.py

```diff
@@ -279,8 +324,27 @@ def predict(self, predict_data: RasaModelData) -> Dict[Text, tf.Tensor]:
         self._training = False  # needed for eager mode
         return self._predict_function(batch_in)
 
-    def save(self, model_file_name: Text) -> None:
-        self.save_weights(model_file_name, save_format="tf")
+    def save(self, model_file_name: Text, overwrite: bool = True) -> None:
+        self.save_weights(model_file_name, overwrite=overwrite, save_format="tf")
+
+    def copy_best(self, model_file_name: Text) -> None:
+        checkpoint_directory, checkpoint_file = os.path.split(self.best_model_file)
+        checkpoint_path = Path(checkpoint_directory)
+
+        # Copy all tf2 model files from the temp location to the final destination
+        for f in checkpoint_path.glob(f"{checkpoint_file}*"):
+            shutil.move(str(f.absolute()), model_file_name + f.suffix)
+
+        # Generate the tf2 checkpoint file, copy+replace to ensure consistency
+        destination_path, destination_file = os.path.split(model_file_name)
+        with open(os.path.join(checkpoint_directory, "checkpoint")) as in_file, open(
```

Comment:
Just for my understanding: What is `os.path.join(checkpoint_directory, "checkpoint")`? When is it created?

Reply:
I think tensorflow creates this file when anything is written to disk. It usually contains some name/path information, e.g. for an NLU model it contains the following:
```
model_checkpoint_path: "component_3_DIETClassifier.tf_model"
all_model_checkpoint_paths: "component_3_DIETClassifier.tf_model"
```

I have noticed that all serialised models have this file, so I've always made sure its copied over. IIRC I've had a failed `rasa test nlu` run because the information in the file wasn't there.